test/mitmproxy/tools/test_main.py

```diff
@@ -1,19 +1,25 @@
+import pytest
+
 from mitmproxy.test import tutils
 from mitmproxy.tools import main
 
 shutdown_script = tutils.test_data.path("mitmproxy/data/addonscripts/shutdown.py")
 
 
-def test_mitmweb():
+@pytest.mark.asyncio
+async def test_mitmweb():
     main.mitmweb([
         "--no-web-open-browser",
         "-q",
+        "-p", "0",
```

Comment:
Why port 0?

Reply:
Port 0 makes the service bind to an arbitrary open port. We don't care where these daemons are listening, we just want the code to run. Asyncio makes synchronising service termination harder in the presence of threads, so this is a quick fix to prevent these two tests from clashing.